### PR TITLE
Chance/v2 updates

### PR DESCRIPTION
**Pull request by ChanceChallacombe**
More tests are needed for the SDK classes both current and added. 

1.  ESMFold posts with sequences longer than allowed have a  nothing string "" returned for  "pdb" and zeroes for the rest of the fields. Is this correct behavior or should it error?
2. Are there AbLang, DNABERT1/2 and ZymCtrl Docs?
3. Caught an API token in one of the examples, it should be retired.
4.  Params validation for the SDK is also needed, maybe there is a way to reuse schemas for all of this after batched payloads are constructed.
5. DNABERT and ProtGPT Fine-tuning should be in their own index as DNABERT 1 and 2 are added

### CODE REVIEW

File: biolmai/cls.py [hunk 1]
```diff
@@ -1,97 +1,176 @@
...
     action_classes = (GenerateAction,)
-    seq_classes = ()
+    generate_input_classes = (AAUnambiguousEmpty(),)
     batch_size = 1
+
+class ProGen2Oas(ProGen2):
+    slug = "progen2-oas"
+
+class ProGen2Medium(ProGen2):
+    slug = "progen2-med"
```
Comment (atqamar) [inline biolmai/cls.py:113]: The ProGen2 Medium slug is `progen2-medium`

File: biolmai/cls.py [hunk 2]
```diff
@@ -1,97 +1,176 @@
...
-    slug = "esm1v_t33_650M_UR90S_5"
-    action_classes = (PredictAction,)
-    seq_classes = (SingleOccurrenceOf("<mask>"), ExtendedAAPlusExtra(extra=["<mask>"]))
-    batch_size = 5
-
+class ESM1vAll(ESM1v):
+    slug = "esm1v-all"
 
 class ESMIF1(APIEndpoint):
     slug = "esmif1"
```
Comment (atqamar) [inline biolmai/cls.py:97]: The ESM-IF1 slug is `esm-if1`

Comment (atqamar) [inline biolmai/cls.py:97]: The ESM-IF1 slug is esm-if1

File: docs/model-docs/progen2/ProGen2_API.rst [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+API Usage
+---------
+
+There are 4 BioLM endpoints corresponding to 4 different ProGen2 models.
+These models are:
+
+* ProGen2 OAS with an endpoint at `https://biolm.ai/api/v2/progen2-oas/ <https://api.biolm.ai/#2424fa3a-144c-4d8e-8391-9e69b9df15d0>`_
+* ProGen2 Medium with an endpoint at `https://biolm.ai/api/v2/progen2-medium/ <https://api.biolm.ai/#06aed4bf-34ac-46da-99d7-791a271c2c53>`_
+* ProGen2 Large with an endpoint at `https://biolm.ai/api/v2/progen2-large/ <https://api.biolm.ai/#4526f1c0-6811-4b4c-9406-8fb774139bc4>`_
+* ProGen2 BFD90 with an endpoint at `https://biolm.ai/api/v2/progen2-bfd90/ <https://api.biolm.ai/#769693ef-1022-4850-8124-06de134d5413>`_
```
Comment (atqamar) [inline docs/model-docs/progen2/ProGen2_API.rst:41]: These API endpoints should have full path of the endpoint, ending in `/generate/`

File: docs/model-docs/esm1v/ESM-1v_API.rst [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+There are 6 BioLM endpoints corresponding to 5 different sized ESM-1v model endpoints and 1 endpoint combining all 5 models.
+These endpoints are:
+
+* ESM-1v model 1 with an endpoint at `https://biolm.ai/api/v2/esm1v-n1/ <https://api.biolm.ai/#ce3145ea-d930-44a8-a468-4d39710381f7>`_
+* ESM-1v model 2 with an endpoint at `https://biolm.ai/api/v2/esm1v-n2/ <https://api.biolm.ai/#abd06778-9836-431a-937b-cfa0479b2632>`_
+* ESM-1v model 3 with an endpoint at `https://biolm.ai/api/v2/esm1v-n3/ <https://api.biolm.ai/#e5e4e266-a42e-455e-8813-3184c170735c>`_
+* ESM-1v model 4 with an endpoint at `https://biolm.ai/api/v2/esm1v-n4/ <https://api.biolm.ai/#2708787e-a1f3-4d3c-aa07-311079c947cc>`_
+* ESM-1v model 5 with an endpoint at `https://biolm.ai/api/v2/esm1v-n5/ <https://api.biolm.ai/#9191c028-bab4-4e7e-b986-0f80d546f6f0>`_
+* ESM-1v all models with an endpoint at `https://biolm.ai/api/v2/esm1v-all/ <https://api.biolm.ai/#67f77d96-c4d8-4a1f-953a-d26330c27315>`_
```
Comment (atqamar) [inline docs/model-docs/esm1v/ESM-1v_API.rst:28]: Endpoint URLs should end with `/predict/`

File: docs/model-docs/proteinfer/ProteInfer_API.rst [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+.. tab-set::
+
+    .. tab-item:: Curl
+        :sync: curl
+
+        .. code:: shell
+
+            curl --location 'https://biolm.ai/api/v1/models/enzyme_function/predict/' \
+            --header 'Content-Type: application/json' \
```
Comment (atqamar) [inline docs/model-docs/proteinfer/ProteInfer_API.rst:45]: Endpoint should be updated to v2 : `v2/proteinfer-ec/predict/`

File: docs/model-docs/proteinfer/ProteInfer_API.rst [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+    .. tab-item:: Python Requests
+        :sync: python
+
+        .. code:: python
+
+            import requests
+            import json
+
+            url = "https://biolm.ai/api/v1/models/enzyme_function/predict/"
```
Comment (atqamar) [inline docs/model-docs/proteinfer/ProteInfer_API.rst:64]: Endpoint should be updated to v2 : `v2/proteinfer-ec/predict/`

File: docs/model-docs/proteinfer/ProteInfer_API.rst [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+            'Authorization' = paste('Token', Sys.getenv('BIOLMAI_TOKEN'))
+            )
+            payload = "{
+                \"items\": [
+                    {
+                        \"sequence\": \"MMQTVLAKIVADKAIWVEARKQQQPLASFQNEVQPSTRHFYDALQGARTAFILECKKASPSKGVIRDDFDPARIAAIYKHYASAISVLTDEKYFQGSFNFLPIVSQIAPQPILCKDFIIDPYQIYLARYYQADACLLMLSVLDDDQYRQLAAVAHSLEMGVLTEVSNEEEQERAIALGAKVVGINNRDLRDLSIDLNRTRELAPKLGHNVTVISESGINTYAQVRELSHFANGFLIGSALMAHDDLHAAVRRVLLGENKVCGLTRGQDAKAAYDAGAIYGGLIFVATSPRCVNVEQAQEVMAAAPLQYVGVFRNHDIADVVDKAKVLSLAAVQLHGNEEQLYIDTLREALPAHVAIWKALSVGETLPAREFQHVDKYVLDNGQGGSGQRFDWSLLNGQSLGNVLLAGGLGADNCVEAAQTGCAGLDFNSAVESQPGIKDARLLASVFQTLRAY\"
+                    }
+                ]
+            }"
+            res <- postForm("https://biolm.ai/api/v1/models/enzyme_function/predict/", .opts=list(postfields = payload, httpheader = headers, followlocation = TRUE), style = "httppost")
```
Comment (atqamar) [inline docs/model-docs/proteinfer/ProteInfer_API.rst:111]: Endpoint should be updated to v2 : `v2/proteinfer-ec/predict/`

**Comment by atqamar** [comment]
@ChanceChallacombe - Excellent first pass at this. I have some comments related to API endpoint names, which should be fixed. We can then merge this in (and leave out remaining/additional changes for a future PR).

File: docs/model-docs/ablang/AbLang_API.rst [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+---------------------------
+Endpoints
+---------------------------
+
+There are 2 BioLM endpoints corresponding to the AbLang Heavy and Light Chain models.
+
+
+* AbLang Heavy Chain endpoint at `https://biolm.ai/api/v2/ablang-heavy/ <https://api.biolm.ai/#1b94f7dc-5dd7-48c6-9944-d933d85bc601>`_
+* AbLang Light Chain endpoint at `https://biolm.ai/api/v2/ablang-light/ <https://api.biolm.ai/#867e99f1-7049-434b-a9a2-d6ff5da0986c>`_
```
Comment (atqamar) [inline docs/model-docs/ablang/AbLang_API.rst:23]: Perhaps we can end these endpoints with `<model-action>` in the URL as a place holder:

`https://biolm.ai/api/v2/ablang-heavy/<model-action>`

File: docs/model-docs/biolmtox/BioLMTox_API.rst [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+
+.. tab-set::
+
+    .. tab-item:: Curl
+        :sync: curl
+
+        .. code:: shell
+
+            curl --location 'https://biolm.ai/api/v1/models/biolmtox_v1/transform/' \
+            --header "Authorization: Token ed3fa24ec0432c5ba812a66d7b8931914c73a208d287af387b97bb3ee4cf907e" \
```
Comment (atqamar) [inline docs/model-docs/biolmtox/BioLMTox_API.rst:42]: Perhaps the Token value should be remove and replaced with a placeholder. Other usages in this PR use an environment variable `$BIOLM_TOKEN`

File: docs/model-docs/biolmtox/BioLMTox_API.rst [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+
+.. tab-set::
+
+    .. tab-item:: Curl
+        :sync: curl
+
+        .. code:: shell
+
+            curl --location 'https://biolm.ai/api/v1/models/biolmtox_v1/predict/' \
+            --header "Authorization: Token ed3fa24ec0432c5ba812a66d7b8931914c73a208d287af387b97bb3ee4cf907e" \
```
Comment (atqamar) [inline docs/model-docs/biolmtox/BioLMTox_API.rst:173]: Same here... (Perhaps the Token value should be remove and replaced with a placeholder. Other usages in this PR use an environment variable `$BIOLM_TOKEN`)

File: docs/model-docs/biolmtox/BioLMTox_API.rst [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+            "instances": [
+               {
+                  "data": {
+                  "text": "MSILVTRPSPAGEELVSRLRTLGQVAWHFPLIEFSPGQQLPQLADQLAALGESDLLFALSQHAVAFAQSQLHQQDRKWPRLPDYFAIGRTTALALHTVSGQKILYPQDREISEVLLQLPELQNIAGKRALILRGNGGRELIGDTLTARGAEVTFCECYQRCAIHYDGAEEAMRWQAREVTMVVVTSGEMLQQLWSLIPQWYREHWLLHCRLLVVSERLAKLARELGWQDIKVADNADNDALLRALQ"
+                  }
+               }
+            ]
+            })
+            headers = {
+            'Authorization': 'Token {}'.format(os.environ['ed3fa24ec0432c5ba812a66d7b8931914c73a208d287af387b97bb3ee4cf907e']),
```
Comment (atqamar) [inline docs/model-docs/biolmtox/BioLMTox_API.rst:202]: Same here... (Perhaps the Token value should be remove and replaced with a placeholder. Other usages in this PR use an environment variable `$BIOLMAI_TOKEN`)

File: docs/model-docs/biolmtox/BioLMTox_API.rst [hunk 12]
```diff
@@ -0,0 +1,293 @@
...
```
Comment (atqamar) [inline docs/model-docs/biolmtox/BioLMTox_API.rst:71]: Same here... (Perhaps the Token value should be remove and replaced with a placeholder. Other usages in this PR use an environment variable `$BIOLMAI_TOKEN`)

**Comment by atqamar** [comment]
Great additions!

- The Travis CI build failures can be fixed by using `from typing import List`, as `list` is not supported in Python 3.8 and prior
- Do remove the Token values from the PR itself

File: docs/model-docs/ablang/AbLang_API.rst (hunk 8 above)
Comment (ChanceChallacombe) [reply docs/model-docs/ablang/AbLang_API.rst:23]: Excellent suggestion!

**Pull request closed without merging**